pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -37,6 +37,18 @@ func GetPipelineFunc(ctx context.Context, k8s kubernetes.Interface, tekton clien
 	cfg := config.FromContextOrDefaults(ctx)
 	pr := pipelineRun.Spec.PipelineRef
 	namespace := pipelineRun.Namespace
+	// if the spec is already in the status, do not try to fetch it again, just use it as source of truth
+	if pipelineRun.Status.PipelineSpec != nil {
```

Comment:
sorry for naive question here, haven't looked at all the changes yet and until I recollect 😜 is it possible to have `nil`  status? do we need an additional check on `status != nil`?

Reply:
`Status PipelineRunStatus `json:"status,omitempty"`` so if it is empty, the struct will be there (no way it can be nil 🙃)